### ISSUE TITLE
Evaluator crashes upon starting

### ISSUE THREAD

**Issue by Etsija** [eclipse-apoapsis/ort-server#1022]
Log from the evaluator suggests the DOS Package Configuration Provider is missing a communications token.  This is curious, since the same token is always used also for the DOS Scanner, which runs successfully. There's been no changes to the Evaluator job configuration.

```
2024-09-12 07:55:39.636	2024-09-12 04:55:39.634 [DefaultDispatcher-worker-1] [traceId=b4f76088-6a12-4e25-913a-db401a7d3381, ortRunId=159, component=evaluator-worker, evaluatorJobId=24] level=INFO  o.e.a.o.t.r.RabbitMqMessageSenderFactory - RabbitMQ queue: 'orchestrator_queue'
2024-09-12 07:55:39.635	2024-09-12 04:55:39.634 [DefaultDispatcher-worker-1] [traceId=b4f76088-6a12-4e25-913a-db401a7d3381, ortRunId=159, component=evaluator-worker, evaluatorJobId=24] level=INFO  o.e.a.o.t.r.RabbitMqMessageSenderFactory - RabbitMQ user: 'admin'
2024-09-12 07:55:39.635	2024-09-12 04:55:39.634 [DefaultDispatcher-worker-1] [traceId=b4f76088-6a12-4e25-913a-db401a7d3381, ortRunId=159, component=evaluator-worker, evaluatorJobId=24] level=INFO  o.e.a.o.t.r.RabbitMqMessageSenderFactory - RabbitMQ server URI: 'amqp://rabbitmq:5672'
2024-09-12 07:55:39.634	2024-09-12 04:55:39.634 [DefaultDispatcher-worker-1] [traceId=b4f76088-6a12-4e25-913a-db401a7d3381, ortRunId=159, component=evaluator-worker, evaluatorJobId=24] level=INFO  o.e.a.o.t.r.RabbitMqMessageSenderFactory - Creating RabbitMQ sender for endpoint 'orchestrator'.
2024-09-12 07:55:39.628	2024-09-12 04:55:39.626 [DefaultDispatcher-worker-1] [traceId=b4f76088-6a12-4e25-913a-db401a7d3381, ortRunId=159, component=evaluator-worker, evaluatorJobId=24] level=INFO  o.e.a.o.t.MessageSenderFactory - Creating a MessageSender of type 'rabbitMQ' for endpoint 'orchestrator'.
2024-09-12 07:55:39.628	2024-09-12 04:55:39.624 [DefaultDispatcher-worker-1] [traceId=b4f76088-6a12-4e25-913a-db401a7d3381, ortRunId=159, component=evaluator-worker, evaluatorJobId=24] level=INFO  o.e.a.o.transport.MessagePublisher - Sending 'EvaluatorWorkerError' message to 'OrchestratorEndpoint'. TraceID: 'b4f76088-6a12-4e25-913a-db401a7d3381'.
2024-09-12 07:55:39.627		at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:702)
2024-09-12 07:55:39.627		at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:715)
2024-09-12 07:55:39.627		at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:811)
2024-09-12 07:55:39.625		at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
2024-09-12 07:55:39.625		at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
2024-09-12 07:55:39.625		at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:99)
2024-09-12 07:55:39.625		at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:111)
2024-09-12 07:55:39.624		at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
2024-09-12 07:55:39.624		at kotlinx.coroutines.JobSupport.makeCompletingOnce$kotlinx_coroutines_core(JobSupport.kt:839)
2024-09-12 07:55:39.624		at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:97)
2024-09-12 07:55:39.624		at kotlinx.coroutines.JobSupport.tryMakeCompleting(JobSupport.kt:867)
2024-09-12 07:55:39.624		at kotlinx.coroutines.JobSupport$ChildCompletion.invoke(JobSupport.kt:1166)
2024-09-12 07:55:39.624		at kotlinx.coroutines.JobSupport.tryFinalizeSimpleState(JobSupport.kt:294)
2024-09-12 07:55:39.624		at kotlinx.coroutines.JobSupport.completeStateFinalization(JobSupport.kt:317)
2024-09-12 07:55:39.624		at kotlinx.coroutines.JobSupport.continueCompleting(JobSupport.kt:947)
2024-09-12 07:55:39.624		at kotlinx.coroutines.JobSupport.access$continueCompleting(JobSupport.kt:22)
2024-09-12 07:55:39.622		at kotlinx.coroutines.DispatchedCoroutine.afterCompletion(Builders.common.kt:251)
2024-09-12 07:55:39.622		at kotlinx.coroutines.DispatchedCoroutine.afterResume(Builders.common.kt:257)
2024-09-12 07:55:39.621		at kotlinx.coroutines.internal.DispatchedContinuationKt.resumeCancellableWith$default(DispatchedContinuation.kt:274)
2024-09-12 07:55:39.621		at kotlinx.coroutines.internal.DispatchedContinuationKt.resumeCancellableWith(DispatchedContinuation.kt:279)
2024-09-12 07:55:39.620		at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
2024-09-12 07:55:39.620		at org.eclipse.apoapsis.ortserver.workers.evaluator.EvaluatorRunner$run$1.invokeSuspend(EvaluatorRunner.kt)
2024-09-12 07:55:39.620		at org.eclipse.apoapsis.ortserver.workers.evaluator.EvaluatorRunner.run(EvaluatorRunner.kt:103)
2024-09-12 07:55:39.620		at org.ossreviewtoolkit.plugins.packageconfigurationproviders.api.PackageConfigurationProviderFactory$Companion.create(PackageConfigurationProviderFactory.kt:52)
2024-09-12 07:55:39.619		at org.ossreviewtoolkit.plugins.packageconfigurationproviders.dos.DosPackageConfigurationProviderFactory.create(DosPackageConfigurationProviderFactory.kt:15)
2024-09-12 07:55:39.619		at org.ossreviewtoolkit.plugins.packageconfigurationproviders.dos.DosPackageConfigurationProviderFactory.create(DosPackageConfigurationProviderFactory.kt:9)
2024-09-12 07:55:39.618	2024-09-12 04:55:39.615 [DefaultDispatcher-worker-1] [traceId=b4f76088-6a12-4e25-913a-db401a7d3381, ortRunId=159, component=evaluator-worker, evaluatorJobId=24] level=ERROR o.e.a.o.w.e.EvaluatorComponent - Evaluator job '24' failed.
2024-09-12 07:55:39.618	java.lang.IllegalStateException: Option token is required but not set.
2024-09-12 07:55:39.612	2024-09-12 04:55:39.613 [DefaultDispatcher-worker-1] [traceId=b4f76088-6a12-4e25-913a-db401a7d3381, ortRunId=159, component=evaluator-worker, evaluatorJobId=24] level=ERROR o.e.a.o.w.evaluator.EvaluatorWorker - Error while processing message with traceId 'b4f76088-6a12-4e25-913a-db401a7d3381': Option token is required but not set.
```

**Comment by Etsija** [comment]
@sschuberth The difference to my yesterday's runs is that ORT has updated to v31 for which I'm reading the changelog now. Are there any breaking changes to plugin config?

**Comment by sschuberth** [comment]
Yes, indeed the new plug-in API is in use for package configuration providers now. However, the DOS provider should have been migrated properly as we upstreamed it a while ago.

**Comment by sschuberth** [comment]
Apparently, ` token`  is marked as a required option to configure the DOS PCP now, but it is not provided by your config.

**Comment by Etsija** [comment]
> Yes, indeed the new plug-in API is in use for package configuration providers now. However, the DOS provider should have been migrated properly as we upstreamed it a while ago.

The curious thing is, yesterday the Evaluator did run and the `ort-result.yml` showed the correct `resolved_configuration` from our backend - the issue was that the new state wasn't being handled properly.

In today's runs, the Evaluator crashes immediately, so naturally there are no resolved configurations seen in `ort-result.yml`. I think the only relevant change is ORT v31, but I am going through all commits since yesterday.

**Comment by Etsija** [comment]
It is, in secrets.properties, just like yesterday.
![Capture](https://github.com/user-attachments/assets/0cf4fa63-4bc8-4ff9-8150-3d677c8bd942)

**Comment by mnonnenmacher** [comment]
This is a bug in ORT, see https://github.com/oss-review-toolkit/ort/pull/9128.

**Comment by Etsija** [comment]
This was fixed with ORT 32.1.0.
